src/org/knotation/draw_io.cljc

```diff
@@ -0,0 +1,127 @@
+(ns org.knotation.draw-io
+  (:require [org.knotation.environment :as en]
+            [org.knotation.rdf :as rdf]
+            [org.knotation.state :as st]
+            [clojure.string :as s]
+            [clojure.data.xml :as xml]))
+
+(def rdfs-label "http://www.w3.org/2000/01/rdf-schema#label")
+(def rdf-type "http://www.w3.org/1999/02/22-rdf-syntax-ns#type")
+(def owl-class "http://www.w3.org/2002/07/owl#Class")
+
+(defn read-xml
+  "Read a draw.io string to a vector of XML objects as maps."
+  [string]
+  (->> string
+       xml/parse-str
+       :content
+       first
+       :content))
+
+(defn get-iris
+  "Given an environment and a vector of XML objects, use any labels to get 
+   existing IRIs. Return a map of the XML node ID to IRI."
+  [env objects]
+  (reduce
+    (fn [m obj]
+      (let [attrs (:attrs obj)]
+        (if (and (contains? attrs :vertex) (contains? attrs :value))
+          (let [id (:id attrs)
+                label (:value attrs)]
+            (if-let [iri (en/name->iri env label)]
+              (assoc m id iri)
+              m))
+          m)))
+    {} objects))
+
+(defn is-class?
+  "Given a map of XML attributes, determine if the node represents a class 
+   based on the styles."
+  [attrs]
+  (let [styles (reduce
+                (fn [m style]
+                  (let [detail (s/split style #"=")]
+                    (into m {(first detail) (second detail)})))
+                {} (s/split (:style attrs) #";"))]
+    (if (contains? styles "rounded")
+      (not (= (get styles "rounded") "0"))
+      false)))
+
+(defn parse-object
+  "Given an XML object, determine if it is a node or an edge. Return a triple 
+   that represents that object or nil."
+  [previous-state iris object]
+  (let [env (st/update-env 
+              (or (::en/env previous-state) en/default-env) 
+                  previous-state)
+        attrs (:attrs object)]
+    (if (contains? attrs :edge)
+      ;; handle an edge
+      (let [subject (or (get iris (:source attrs)) (str "_:" (:source attrs)))
+            target (or (get iris (:target attrs)) (str "_:" (:target attrs)))
+            property (en/name->iri env (:value attrs))]
```

Comment:
We need to handle the case where the property name is not found.

Reply:
Would this be an error or would it create a new property?

Reply:
It would be an error, like "Unknown predicate 'foo'"